lib/requirements.txt

```diff
@@ -2,3 +2,4 @@ flake8==3.8.0
 ansible-core==2.11.3
 ansible-lint==5.1.2
 nose==1.3.7
+azure==4.0.0
```

Comment:
What problem were you seeing that this fixed?  Curious if a muchos command was failing and if so what error you saw.

Reply:
@zachradtka, it would also be good to know a bit more about the environment where the above issue was detected - the GitHub Actions runner seemed to not require this installed, but that may very well be because other installations (like the Azure CLI) might have provided that package.

Going out on a proverbial limb - originally, we used to install the `ansible[azure]` package, which used to install the required parts of the `azure` package implicitly. Earlier this year when we switched to more current Ansible (core) a possible gap might have been created, which (I'm hypothesizing here) may not come up in the CI (or in local environments) where AZ CLI was also installed.

Reply:
If you want the exact error, I can re-run through my install steps again.

Reply:
> If you want the exact error, I can re-run through my install steps again.

Thanks, but I don't need that.  Your description was very helpful.

Reply:
> I was installing on an "existing" cluster.

So you are not using Muchos to provision a cluster (in AWS or Azure), just trying to use it provision software on a cluster you already have?  Sorry to restate, I think that is what you mean, just want to make sure I understand your problem.

Reply:
That is correct. I already have the cluster installed and was not using muchos to provision hardware.

Reply:
@zachradtka - did you run [https://github.com/apache/fluo-muchos/blob/main/scripts/install-ansible-for-azure](./scripts/install-ansible-for-azure)?

That is the recommended way to ensure all the dependencies, including Azure modules for Ansible, are pulled in. That process implicitly installs the `azure` package.

If this was not run, I'd recommend updating the README or such if we have not mentioned this explicitly, but I'd not want to add the azure package to the top-level requirements.txt.

Reply:
> @zachradtka - did you run https://github.com/apache/fluo-muchos/blob/main/scripts/install-ansible-for-azure?

@arvindshmicrosoft  should this have to be run even when someone is not using Azure?

Reply:
@keith-turner I did not run https://github.com/apache/fluo-muchos/blob/main/scripts/install-ansible-for-azure prior to running `muchos setup`.

Reply:
It is mentioned to run `install-ansible-for-azure` in the readme: https://github.com/apache/fluo-muchos/blob/main/README.md?plain=1#L73-L74, but it is in the Azure subsection. I can move it up to the "Common" subsection. Would that be the correct place to put it?

Reply:
Oh wait - I missed the part about you running on an `existing` cluster. It should not then need the `azure` package at all, and my steps about the Azure pre-requisites should not be needed.

@zachradtka - if it is not too much to ask, it would be really helpful to get the original error to diagnose what is going on here. Clearly, some Azure specific module was invoked when it should not have been invoked.

Reply:
I'm guessing the error is this:
```
Traceback (most recent call last):
  File "/home/centos/work/fluo-muchos/lib/main.py", line 22, in <module>
    from muchos.config import DeployConfig
  File "/home/centos/work/fluo-muchos/lib/muchos/config/__init__.py", line 20, in <module>
    from .azure import AzureDeployConfig
  File "/home/centos/work/fluo-muchos/lib/muchos/config/azure.py", line 23, in <module>
    from .azurevalidations import validate_azure_configs
  File "/home/centos/work/fluo-muchos/lib/muchos/config/azurevalidations.py", line 24, in <module>
    from azure.mgmt.compute import ComputeManagementClient
ModuleNotFoundError: No module named 'azure'
```

If so, the reason is [this](https://github.com/apache/fluo-muchos/blob/7decbf3069e592414cfd184b98e424a503c09b6e/lib/muchos/config/__init__.py#L19-L20)

We'll need to think through how to not load the non-required libraries. 

Meanwhile, for this PR, my suggestion remains the same - let's not add the reference to the `azure` library when it is not required (i.e. don't add it in requirements.txt). I suggest keeping this PR specific to the Hadoop checksum change, and we can work on the changes / refactoring needed to not require the azure library, separately.

Reply:
Yup, that is the issue. Sorry I didn't have a chance to run this today. I can run it tomorrow just to confirm.

Reply:
NP - I am including a patch below which will address the "importing" issue. Feel free to leverage the same in this PR if you like, or LMK if you'd prefer that I raise another PR with the below changes. In either case, you can reset your change to requirements.txt.

```
diff --git a/lib/muchos/config/__init__.py b/lib/muchos/config/__init__.py
index 99a88b5..27ba8f4 100644
--- a/lib/muchos/config/__init__.py
+++ b/lib/muchos/config/__init__.py
@@ -15,10 +15,6 @@
 # limitations under the License.
 #

-from .existing import ExistingDeployConfig
-from .ec2 import Ec2DeployConfig
-from .azure import AzureDeployConfig
-
 from configparser import ConfigParser


@@ -35,6 +31,8 @@ def DeployConfig(
     cluster_type = c.get("general", "cluster_type")

     if cluster_type == "existing":
+        from .existing import ExistingDeployConfig
+
         return ExistingDeployConfig(
             deploy_path,
             config_path,
@@ -45,6 +43,8 @@ def DeployConfig(
         )

     if cluster_type == "ec2":
+        from .ec2 import Ec2DeployConfig
+
         return Ec2DeployConfig(
             deploy_path,
             config_path,
@@ -55,6 +55,8 @@ def DeployConfig(
         )

     if cluster_type == "azure":
+        from .azure import AzureDeployConfig
+
         return AzureDeployConfig(
             deploy_path,
             config_path,
diff --git a/lib/tests/azure/test_config.py b/lib/tests/azure/test_config.py
index 69469a3..17ecdc7 100644
--- a/lib/tests/azure/test_config.py
+++ b/lib/tests/azure/test_config.py
@@ -15,7 +15,7 @@
 # limitations under the License.
 #

-from muchos.config import AzureDeployConfig
+from muchos.config.azure import AzureDeployConfig


 def test_azure_cluster():
diff --git a/lib/tests/ec2/test_config.py b/lib/tests/ec2/test_config.py
index 58f1afe..4b620ac 100644
--- a/lib/tests/ec2/test_config.py
+++ b/lib/tests/ec2/test_config.py
@@ -15,7 +15,7 @@
 # limitations under the License.
 #

-from muchos.config import Ec2DeployConfig
+from muchos.config.ec2 import Ec2DeployConfig


 def test_ec2_cluster():
diff --git a/lib/tests/existing/test_config.py b/lib/tests/existing/test_config.py
index 1be2a86..4e2f3d6 100644
--- a/lib/tests/existing/test_config.py
+++ b/lib/tests/existing/test_config.py
@@ -15,7 +15,7 @@
 # limitations under the License.
 #

-from muchos.config import ExistingDeployConfig
+from muchos.config.existing import ExistingDeployConfig
```

Reply:
I just opened a bug for this https://github.com/apache/fluo-muchos/issues/416

Reply:
It seems like the bug fix could be a separate change from hadoop support. Currently #416 is a blocker preventing running of the latest Muchos.

Reply:
I have updated this PR to just include support for Hadoop 2.10.1

Reply:
Thanks - I raised #417 to address the `azure` package issue.

Reply:
@keith-turner if you are good with the PR, would you like to resolve these 2 comments? I have approved the PR, with a second vote, we can merge.